### PR TITLE
Add Verb to dependency name

### PR DESCRIPTION
**Pull request by abaranch**
Only for StatusMonitor

### CODE REVIEW

**Comment by tokaplan** [comment]
Host name will remain in the Name of HTTP dependencies, right? QuickPulse processor relies on that to filter out requests to QPS.

File: Src/DependencyCollector/Shared/Implementation/ProfilerHttpProcessing.cs [hunk 1]
```diff
@@ -205,17 +222,27 @@ private object OnBegin(object thisObj, bool isAsyncCall)
...
-                string resourceName = this.GetResourceName(thisObj);
+                string url = this.GetUrl(thisObj);
+                string httMethod = this.GetMethod(thisObj);
+
+                string resourceName = url;
+
+                if (!string.IsNullOrEmpty(httMethod))
+                {
+                    // We add prefix that we will remove on the backend
+                    // We need prefix because url can contain spaces and we need better separator to distinguish from previous versions
```
Comment (Dmitry-Matveev) [inline Src/DependencyCollector/Shared/Implementation/ProfilerHttpProcessing.cs]: Minor: Seems to be an obsolete comment.

File: Src/DependencyCollector/Shared/Implementation/ProfilerHttpProcessing.cs [hunk 2]
```diff
@@ -173,22 +173,39 @@ public void OnExceptionForEndGetRequestStream(object context, object exception,
...
+        /// <returns>The url if possible otherwise empty string.</returns>
+        internal string GetUrl(object thisObj)
         {
             WebRequest webRequest = thisObj as WebRequest;
             string resource = string.Empty;
             if (webRequest != null && webRequest.RequestUri != null)
             {
                 resource = webRequest.RequestUri.ToString();
-            }
+            }
```
Comment (dnduffy) [inline Src/DependencyCollector/Shared/Implementation/ProfilerHttpProcessing.cs]: Extra trailing spaces.

File: Test/DependencyCollector/FunctionalTests/FuncTest/RDDTests.cs [hunk 3]
```diff
@@ -1019,7 +1039,7 @@ private void ExecuteAzureSDKTests(TestWebApplication testWebApplication, int cou
         private void ValidateRddTelemetryValues(TelemetryItem<RemoteDependencyData> itemToValidate, string remoteDependencyNameExpected, string commandNameExpected, int countExpected, TimeSpan accessTimeMax, bool successFlagExpected, bool asyncFlagExpected)
         {
             DependencySourceType source = sourceExpected;
-            Assert.IsTrue(itemToValidate.Data.BaseData.Name.Contains(remoteDependencyNameExpected), "The remote dependancy name is incorrect");
+            Assert.IsTrue(itemToValidate.Data.BaseData.Name.Contains(remoteDependencyNameExpected), "The remote dependancy name is incorrect: " + remoteDependencyNameExpected + "Collected " + itemToValidate.Data.BaseData.Name);
```
Comment (Dmitry-Matveev) [inline Test/DependencyCollector/FunctionalTests/FuncTest/RDDTests.cs]: Minor: You might need a space before "Collected  " as well to better separate expected name.

File: Src/DependencyCollector/Shared/Implementation/ProfilerHttpProcessing.cs [hunk 4]
```diff
@@ -173,22 +173,39 @@ public void OnExceptionForEndGetRequestStream(object context, object exception,
...
             return resource;
         }
-          
+
+        /// <summary>
+        /// Gets HTTP request Verb.
+        /// </summary>
+        /// <param name="thisObj">Represents web request.</param>
+        /// <returns>The Verb if possible otherwise empty string.</returns>
+        private string GetMethod(object thisObj)
```
Comment (dnduffy) [inline Src/DependencyCollector/Shared/Implementation/ProfilerHttpProcessing.cs]: Why "object" instead of "WebRequest"?

**Comment by Dmitry-Matveev** [comment]
:shipit:

**Comment by dnduffy** [comment]
:+1:

**Comment by abaranch** [comment]
@tokaplan Alex, we are adding prefix. Does that affect QuickPulse?

**Pull request merged**